packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -66,13 +106,40 @@ class MenuContent extends React.Component {
     return { isSelectableMenu: this.isSelectable() };
   }
 
+  componentWillReceiveProps(newProps) {
+    if (newProps.isFocused) {
+      this.needsFocus = this.needsFocus || this.props.isFocused !== newProps.isFocused;
+    } else {
+      this.needsFocus = false;
+    }
+  }
+
+  componentDidUpdate() {
+    this.validateFocus(this.contentNode);
+  }
+
   onKeyDownBackButton(event) {
     if (event.nativeEvent.keyCode === MenuUtils.KEYCODES.ENTER || event.nativeEvent.keyCode === MenuUtils.KEYCODES.SPACE) {
       event.preventDefault();
       this.props.onRequestBack();
     }
   }
 
+  validateFocus(node) {
+    if (this.needsFocus && node) {
+      node.focus();
+      this.needsFocus = document.activeElement !== node;
```

Comment:
when would this ever not be false?
the previous line you set focus on the node making it the active element correct?

Reply:
If the item cannot take focus as a result of being hidden or overlapped, potentially as the result of animation or the timing of the callback. Largely its to guard against the behavior of focus trap, when it sets a list of tabbable items, if for any reason the item is removed from that list, it won't receive focus.